lua/telescope/entry_manager.lua

```diff
@@ -155,7 +155,10 @@ function EntryManager:add_entry(picker, score, entry)
     info.looped = info.looped + 1
 
     if container[2] > score then
-      -- print("Inserting: ", picker, index, node, new_container)
+      return self:_insert_container_before(picker, index, node, new_container)
+    end
+
+    if score < 1 and container[2] == score and #entry.ordinal < #container[1].ordinal then
```

Comment:
Would be wrapping `#entry.ordinal < #container[1].ordinal` in a `break_ties(current_entry, existing_entry)` function that users can override and pass via `opts` to `pickers:new()` something we want? Which defaults to preferring shorter ordinals, of course.

Would address
#1080 (I guess that's what OP wanted)
#939 (I think)

and I'm sure allows for cool customization of other stuff down the line.

Reply:
Yeah, let's do that in a follow up PR